### ISSUE TITLE
`piecewise_interpolate_schedule` sometimes generate nan learning rate if the number of transition steps is low.

### ISSUE THREAD

**Issue by VRichardJP** [google-deepmind/optax#1403]
For example:

```python
import optax

sched = optax.cosine_onecycle_schedule(
    transition_steps=3,
    peak_value=0.03668598220619202,
    pct_start=0.19848356535253986,
    div_factor=4480.102412853588,
    final_div_factor=3598.8465977101146,
)

for step in range(3):
    print(f"Step {step}: {sched(step):.6f}")

```

Output:
```
Step 0: nan
Step 1: nan
Step 2: nan
```

This is caused by 0 appearing in the boundaries, leading to `interval_sizes[0] == 0`

```
# locals() of piecewise_interpolate_schedule
{
    'interpolate_type': 'cosine',
    'init_value': 8.188648121288145e-06,
    'boundaries_and_scales': {0: 4480.102412853588, 3: 6.202242220930262e-08},
    'boundaries': (0, 3),
    'scales': (4480.102412853588, 6.202242220930262e-08),
    'bounds': array([0, 0, 3]),
    'interpolate_fn': <function _cosine_interpolate at 0x7f487f325f80>,
    'interval_sizes': array([0, 3]),
    'values': array([8.18864812e-06, 3.66859822e-02, 2.27535348e-09])
}
```

Note that a low number of transition steps will typically occur in unit tests (e.g. testing a training script with just a few items).

**Comment by rdyro** [comment]
Good catch, I'll attempt to merge a fix soon, thanks for reporting!

**Comment by rdyro** [comment]
This should be fixed now, thanks for reporting! Feel free to re-open if you find any issues after the fix.
